### PR TITLE
[U2] [5.11.0] Fix Authentication with SAAS app in super-tenant using another tenant's user with SMS-OTP as 2nd factor

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
Related issue: https://github.com/wso2/product-is/issues/14965

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 1]
```diff
@@ -1639,4 +1638,26 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
                     username, e);
         }
     }
+
+    /**
+     * Returns AuthenticatedUser object from context.
+     *
+     * @param context AuthenticationContext.
+     * @return AuthenticatedUser
```
Comment (AnuradhaSK) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1646]:
```suggestion
     * @return AuthenticatedUser.
```

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java [hunk 2]
```diff
@@ -1639,4 +1638,26 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
...
+     */
+    private AuthenticatedUser getAuthenticatedUser(AuthenticationContext context) {
+
+        AuthenticatedUser authenticatedUser = null;
+        Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
+        for (StepConfig stepConfig : stepConfigMap.values()) {
+            AuthenticatedUser authenticatedUserInStepConfig = stepConfig.getAuthenticatedUser();
+            if (stepConfig.isSubjectAttributeStep() && authenticatedUserInStepConfig != null) {
+                // Make a copy of the user from the subject attribute step as we might modify this within
+                // the authenticator.
```
Comment (AnuradhaSK) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java:1656]: Make this multiline comment

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 1 above)
Comment (ZiyamSanthosh) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/154/commits/a67fa057a7d8e1efb3b52117cad0f7fb71f7205b

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java (hunk 2 above)
Comment (ZiyamSanthosh) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java]: Fixed in https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/154/commits/a67fa057a7d8e1efb3b52117cad0f7fb71f7205b

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**